src/com/google/javascript/jscomp/CommandLineRunner.java

```diff
@@ -642,6 +599,28 @@
         usage = "Prevent injecting the named runtime libraries.")
     private List<String> noinjectLibrary = new ArrayList<>();
 
+    @Option(name = "--dependency_mode",
+        hidden = true,
+        usage = "Specifies how the compiler should determine the set and order "
+            + "of files for a compilation. Options: NONE the compiler will include all "
+            + "src files in the order listed, STRICT files will be included and sorted by "
+            + "starting from namespaces or files listed by the --entry_point flag - files "
```

Comment:
An interesting question with dependency management is the *order* in which files get included. This might need some documentation - as a user I'd expect the effective order to be a pre-order traversal of the dependency tree. Is that what happens here? Maybe spell it out, possible even with an example somewhere?